platform/ios/src/MGLMapView.mm

```diff
@@ -4064,9 +4051,15 @@ - (MGLAnnotationTag)annotationTagAtPoint:(CGPoint)point persistingResults:(BOOL)
                 {
                     return true;
                 }
-                
+
                 CGPoint calloutAnchorPoint = MGLPointRounded([self convertCoordinate:annotation.coordinate toPointToView:self]);
                 CGRect frame = CGRectInset({ calloutAnchorPoint, CGSizeZero }, -CGRectGetWidth(annotationView.frame) / 2, -CGRectGetHeight(annotationView.frame) / 2);
+
+                // We need to take any offset into consideration. Note that a large offset will result in a
+                // large value for `_unionedAnnotationRepresentationSize` - which will affect performance for lots
+                // of annotations. Keep the offset as small as possible.
```

Comment:
Would this note be worth adding to the API docs for [`centerOffset`](https://github.com/mapbox/mapbox-gl-native/blob/ios-v4.1.0/platform/ios/src/MGLAnnotationView.h#L158)?